src/components/UserSession/index.tsx

```diff
@@ -0,0 +1,15 @@
+import constate from 'constate';
+import { useState } from 'react';
+import { LoggedInUserFragment } from '../../api';
+
+function useUserSession() {
+  const [userSession, setUserSession] = useState<LoggedInUserFragment | null>(
+    null
+  );
+
+  return { userSession, setUserSession };
```

Comment:
Let's move the effect to load the session here. We'll never not want to load the current session, even if it comes from a cache in the future.

Let's also change the return type
```
[userSession, loading, setter]
```
That makes it easier to use a different var name in components. also we'll need the loading boolean.

Reply:
Is it necessary to send back `setter` since we'll be moving the session query here?

Reply:
Yeah unfortunately. On login/logout. Unless that can be automatically connected to via apollo client listener?

Reply:
Err nevermind, we'll need the setter for post-login

Reply:
running into a couple of issues with returning arrays for the return type:

1. because we're going with array return type, there will be times when we needlessly import the `session` and `loading` property even when all we need is the setter 

2. for some reason, constate is returning weird typings for the `setter`:
```typescript
const [_userSession, _loading, setUserSession] = useSession();

// See error output from this line:
setUserSession(data.login.user);

^ const setUserSession: boolean | LoggedInUserFragment | React.Dispatch<React.SetStateAction<LoggedInUserFragment | null>> | null
This expression is not callable.
  Not all constituents of type 'boolean | LoggedInUserFragment | Dispatch<SetStateAction<LoggedInUserFragment | null>>' are callable.
    Type 'false' has no call signatures.ts(2349)
```

Reply:
I just pushed up a version without array types, but instead an object type as a return type. This seems cleaner IMO - let me know what you think.

Reply:
It's actually TS trying to infer the type based on the array items. It thinks you want "an array with items that are of any of these types".

Instead if you declare the array "as a constant", `[] as const`. then it knows you want a tuple instead.

Reply:
`as const` did it. But our linter isn't too happy about unused values. So at the moment this:
```typescript
const { setUserSession } = useSession();
```

has become this:
```typescript
  const [_userSession, _loading, setUserSession] = useSession();
```

Cool if I disable `@typescript-eslint/no-unused-vars` for this line?